File: bots/python/randobot.py
import time
import pprint
import numpy as np
import os
import sys

# Add parent directory to path for api import
sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
import apis.codemarket as cm

# Init stuff
ledger = {}
name = str(np.random.randint(100000))
pp = pprint.PrettyPrinter(indent=4)

# Loop indefinitely and perform random actions
uuid = cm.register_vendor(name)['uuid']
while 'error' not in ledger.keys():
    ledger = cm.get_ledger_state(uuid)
    me = ledger[name]
    to_move = np.random.randint(len(me[2])) # Choose random item
    store = ledger['stored'][2][to_move]
    stock = me[2][to_move]

    # Positive for stocking, negative for storing
    if store > stock:
        to_stock = np.random.randint(store) + 1
    else:
        to_stock = -(np.random.randint(stock) + 1)

    # Create the stock request with a random price
    cm.stock(me[0][to_move], np.random.rand(), to_stock, uuid)
    time.sleep(10)

pp.pprint(ledger)

File: bots/javascript/randobot.js
const cm = require('../apis/codemarket');

class RandoBot {
    async run() {
        let ledger = {};
        let name = Math.floor(Math.random() * 100000).toString();
        let registration = await cm.register_vendor(name);
        let uuid = registration.uuid;
        while (!('error' in ledger)) {
            ledger = await cm.get_ledger_state(uuid);
            let me = ledger[name];
            let to_move = Math.floor(Math.random() * me[2].length);
            let store = ledger.stored[2][to_move];
            let stock = me[2][to_move];
        
            let to_stock = 0
            if (store > stock) {
                to_stock = Math.floor(Math.random() * store);
            } else {
                to_stock = Math.floor(Math.random() * stock);
            }
        
            await cm.stock(me[0][to_move], Math.random(), to_stock, uuid);
            await new Promise(r => setTimeout(r, 10000));
        }
    }
}

let r = new RandoBot();
r.run()
 .then(console.log(r.ledger))
 .catch((e) => console.log(e));